atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTest.java

```diff
@@ -220,6 +223,16 @@ public void testStoreByteStreamFiveMegaBytes_compressedStream_incompressible() t
         storeAndCheckByteStreams(compressedStore, getIncompressibleBytes(5_000_000));
     }
 
+    @Test
+    public void testStoreToMaxMemStream() throws IOException {
+        storeAndCheckByteStreams(maxMemStore, getIncompressibleBytes(100));
+    }
+
+    @Test
+    public void testStoreHugeToMaxMemStream() throws IOException {
+        storeAndCheckByteStreams(maxMemStore, getIncompressibleBytes(20_000_000));
```

Comment:
"fence post" tests for this exact thing doesn't make sense. This test itself exposes a bug in the current stream logic (if you don't count the fix included in this commit).  There are already other tests that include testing block boundaries (for the case where the in-memory size is smaller)